CONTRIBUTING.md

```diff
@@ -0,0 +1,129 @@
+what your template does.
+1. **Save the template**. Hit the "Save" button. The output of the template
+applied to the current example will appear in the right sidebar.
+1. **Upload the template(s).** Submit a PR using the instructions
+[here](#uploading-templates).
+
+## Getting Started Using Jinja to Write Templates
+Here is a quick crash course on using [Jinja](https://jinja.palletsprojects.com/en/3.0.x/)
+to write templates. More advanced usage is in the [cookbook](#jinja-cookbook).
+
+Generally in a template, you'll want to use a mix of hard-coded data that is
+task-specific and stays the same across examples, and commands that tailor the
+prompt and output to a specific example.
+
+To write text that should be rendered as written, just write it normally. The
+following "template" will produce the same text every time:
+```jinja2
+This is just literal text that will be printed the same way every time. 
+```
+
+To make your template do something more interesting, you'll need to use Jinja
+expressions. Jinja expressions are surrounded by curly braces `{` and `}`.
+One common thing you'll want to do is access information in the dataset example.
+When applied to an example, you can access any value in the example dictionary
+via its key. If you just want to print that value surround it in double curly
+braces. For example, if you want to print a value with the key `text`, use this:
+```jinja2
+The text in this example is {{ text }}.
+```
+
+You can also use information from the example to control behavior. For example,
+suppose we have a label with the key `label` in our example, which either has a
+value of 0 or 1. That's not very "natural" language, so maybe we want to decide
+which label name to use based on the example. We can do this by creating a list
+and indexing it with the example key:
+```jinja2
+The label for this example is {{ ["Label A", "Label B"][label] }}.
+```
+We can also use dictionaries for the same thing:
+```jinja2
+The label for this example is {{
+{"a": "Label A",
+ "b": "Label B"
+}[label]
+}}.
+```
+
+Note that some things in a template are particular to the task, and should not be
+modified by downstream steps that try to increase the diversity of the prompts.
+A common example is listing label names in the prompt to provide choices. Anything
+that should not be modified by data augmentation should be surrounded by double
+curly braces and quoted. For example:
+```jinja2
+The choices are {{"a"}}, {{"b"}}, and {{"c"}}.
+```
+
+Finally, remember that a template must produce two strings: a prompt and an output.
+To separate these two pieces, use three vertical bars `|||`.
+So, a complete template for AG News could be:
+```jinja2
+{{text}} 
+Is this a piece of news regarding {{"world politics"}}, {{"sports"}}, {{"business"}}, {{"science and technology"}}? ||| 
+{{ ["World politics", "Sports", "Business", "Science and technology"][label] }}
+```
+   
```

Comment:
@VictorSanh and I were thinking that blank templates should just get filtered automatically.  So for Trec you could write: 

{% if label_coarse == 0 %}
Is this question asking for {{"a definition, a description, a manner of action, or a reason"}}? {{text}}
||| ... 
{% endif %}

Think that is nicer than having all the templates in one.

Reply:
Concrete example: https://github.com/bigscience-workshop/promptsource/blob/main/templates/trec/templates.yaml#L37 would be broken down into 4 templates which can optionally return "blank" in which case, we should ignore them when prompting the dataset

Reply:
Ah yes, this is an elegant solution to #4 . 👍